pkg/distributor/distributor.go

```diff
@@ -270,10 +274,18 @@ func New(cfg Config, clientConfig ingester_client.Config, limits *validation.Ove
 	subservices := []services.Service(nil)
 	subservices = append(subservices, haTracker)
 
+	var requestBufferPool util.Pool
+	if cfg.MaxRequestPoolBufferSize > 0 {
+		requestBufferPool = util.NewBucketedBufferPool(1e3, cfg.MaxRequestPoolBufferSize, 2)
```

Comment:
Since these are bytes, and scale in power of two, I'd start with a power of two.
```suggestion
		requestBufferPool = util.NewBucketedBufferPool(1<<10, cfg.MaxRequestPoolBufferSize, 2)
```